src/doc/rustc/src/codegen-options/index.md

```diff
@@ -244,18 +244,35 @@ flag for another way to specify the linker.
 
 ## linker-features
 
-This flag allows enabling or disabling specific features used during linking.
+The `-Clinker-features` flag allows enabling or disabling specific features used during linking.
 
-The flag accepts a comma-separated list of features, individually enabled (`+feature`) or disabled
-(`-feature`).
+These feature flags are a flexible extension mechanism that is complementary to linker flavors,
+designed to avoid the combinatorial explosion of having to create a new set of flavors for each
+linker feature we'd want to use.
 
-Currently, only one such feature is stable:
-- `lld`: toggles the usage of the lld linker, either the system-installed binary, or the self-contained
-  `rust-lld` linker.
+The flag accepts a comma-separated list of features, individually enabled (`+features`) or disabled
+(`-features`).
 
-If you want to opt out of the usage of the `lld` linker (for targets where it is configured as the default linker),
+Currently only one is stable, and only on the `x86_64-unknown-linux-gnu` target:
+- `lld`: to toggle using the lld linker, either the system-installed binary, or the self-contained
+  `rust-lld` linker (via the `-Clink-self-contained=+linker` flag).
+
+To opt out of using the `lld` linker (for targets where it is configured as the default linker),
 use `-Clinker-features=-lld`.
 
+In general, this design allows:
```

Comment:
This sounds a bit more like an internal comment than user-facing documentation.

Reply:
It's the unstable book documentation but ok I've removed it.